src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -507,12 +501,20 @@ private void check() {
                 return;
             }
             recurrencePeriod = 0;
+            handleExit(exitCode, () -> output);
+        }
+
+        @FunctionalInterface
+        private interface OutputSupplier {
+            byte[] produce() throws IOException, InterruptedException;
+        }
+        private void handleExit(int exitCode, OutputSupplier output) throws IOException, InterruptedException {
             Throwable originalCause = causeOfStoppage;
             if ((returnStatus && originalCause == null) || exitCode == 0) {
-                getContext().onSuccess(returnStatus ? exitCode : returnStdout ? new String(output, StandardCharsets.UTF_8) : null);
+                getContext().onSuccess(returnStatus ? exitCode : returnStdout ? new String(output.produce(), StandardCharsets.UTF_8) : null);
```

Comment:
:ant: Nested ternary operators are super hard to read for most developers without line breaks between them

Reply:
Perhaps. Anyway that predates this PR.